invenio_rdm_records/tokens/permissions.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Permissions for Resource Access Token."""
+
+from collections import namedtuple
+
+RATNeed = namedtuple(
+    "RATNeed", ["method", "signer_id", "pid_value", "file_key", "permission"]
+)
+"""
+A Resource Access Tokens Need
+ex: RATNeed("files", "3", "byy8e-7yz93", "file5.png", "edit")
+"""
```

Comment:
```suggestion

_Need = namedtuple("Need", ["method", "signer_id", "pid_value", "file_key", "permission"]

RATNeed = partial(_Need, "rat")
"""
A Resource Access Tokens Need
ex: RATNeed("3", "byy8e-7yz93", "file5.png", "write_file")
"""
```

minor: two changes:

- Define a "base" `_Need` as done e.g. [for communtiies](https://github.com/inveniosoftware/invenio-communities/blob/49b69d4483db3a56c68a40dfb294966403512c45/invenio_communities/generators.py#L27-L30).
- the `permission` value can be a bit more explicit. imagine this could be used for all sorts of other resources/granularity so you could have values `read_record` for only viewing metadata, `write_file` for only writing a single file, `read` to read record + files. No need to change the current generators though (see relevant comment).

Don't worry too much about the Zenodo legacy values that are in the RFC/codebase, we can easily configure these to match things in legacy.